tools/generate-sponsors.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * @fileoverview
+ * This script generates a markdown file containing sponsor information.
+ * It fetches sponsor data from specified JSON files, formats
+ * the data into HTML sections, and writes the output to a new markdown file.
+ *
+ * Usage:
+ *   Run the script using Node.js:
+ *   node tools/generate-sponsors.js
+ *
+ * Author: Amaresh S M
+ */
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+const fs = require("node:fs").promises;
+const path = require("node:path");
+const { stripIndents } = require("common-tags");
+
+//-----------------------------------------------------------------------------
+// Data
+//-----------------------------------------------------------------------------
+
+const SPONSORS_FILE_PATH = path.resolve(
+	__dirname,
+	"../src/_data/sponsors.json",
+);
+const TECH_SPONSORS_FILE_PATH = path.resolve(
+	__dirname,
+	"../src/_data/techsponsors.json",
+);
+const NEW_FILE_PATH = path.resolve(__dirname, "../includes/sponsors.md");
+
+const TECH_SPONSORS_IMAGE_PATH =
+	"https://raw.githubusercontent.com/eslint/eslint.org/main/src";
+
+const HEIGHTS = {
+	diamond: 160,
```

Comment:
We're keeping diamond logos the same size as platinum

```suggestion
	diamond: 128,
```